element_volume/volume.py

```diff
@@ -0,0 +1,261 @@
+                )
+        else:
+            if {"paramset_idx": paramset_idx} in cls.proj():
+                raise dj.DataJointError(
+                    f"The specified paramset_idx {paramset_idx} already exists,"
+                    f" please pick a different one."
+                )
+            cls.insert1(param_dict)
+
+
+@schema
+class SegmentationTask(dj.Manual):
+    definition = """
+    -> Volume
+    -> SegmentationParamset
+    ---
+    segmentation_output_dir='': varchar(255)  #  Output directory of the segmented results relative to root data directory
+    task_mode='load': enum('load', 'trigger')
+    """
+
+
+@schema
+class Segmentation(dj.Computed):
+    definition = """
+    -> SegmentationTask
+    """
+
+    class Mask(dj.Part):
+        definition = """ # A mask produced by segmentation.
+        -> master
+        mask            : smallint
+        ---
+        mask_npix       : int       # number of pixels in ROIs
+        mask_center_x   : float     # center x coordinate in pixel
+        mask_center_y   : float     # center y coordinate in pixel
+        mask_center_z   : float     # center z coordinate in pixel
+        mask_xpix       : longblob  # x coordinates in pixels
+        mask_ypix       : longblob  # y coordinates in pixels
+        mask_zpix       : longblob  # z coordinates in pixels
+        mask_weights    : longblob  # weights of the mask at the indices above
+        """
+
+    def make(self, key):
+        # NOTE: convert seg data to unit8 instead of uint64
+        task_mode, seg_method, output_dir, params = (SegmentationTask * SegmentationParamset & key).fetch1(
+            "task_mode", "segmentation_method", "processing_output_dir", "params"
+        )
+        output_dir = find_full_path(get_volume_root_data_dir(), output_dir).as_posix()
+        if task_mode == "trigger" and seg_method.lower() == "cellpose":
+            from cellpose import models as cellpose_models
+            volume_data = (Volume & key).fetch1("volume")
+            model = cellpose_models.CellposeModel(model_type=params['model_type'])
+            cellpose_results = model.eval(
+                [volume_data],
+                diameter=params['diameter'],
+                channels=params.get('channels', [[0, 0]]),
+                min_size=params['min_size'],
+                z_axis=0,
+                do_3D=params['do_3d'],
+                anisotropy=params['anisotropy'],
+                progress=True,
+            )
+            masks, flows, styles = cellpose_results
+
+            mask_entries = []
+            for mask_id in set(masks[0].flatten()) - {0}:
+                mask = np.argwhere(masks[0] == mask_id)
+                mask_zpix, mask_ypix, mask_xpix = mask.T
+                mask_npix = mask.shape[0]
+                mask_center_z, mask_center_y, mask_center_x = mask.mean(axis=0)
+                mask_weights = np.full_like(mask_zpix, 1)
```

Comment:
Cellpose returns the mask weights. They are stored in `flows` (`https://cellpose.readthedocs.io/en/latest/outputs.html`).

Reply:
Will implement this in next PR